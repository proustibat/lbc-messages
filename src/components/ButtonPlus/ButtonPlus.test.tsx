import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { ButtonPlus, ButtonPlusProps } from './index';

const props: ButtonPlusProps = {
  onClick: jest.fn()
};
describe('ButtonPlus', () => {
  afterEach(() => {
    cleanup();
    jest.resetAllMocks();
  });

  it('should render correctly by default', () => {
    // Given / When
    const { container } = render(<ButtonPlus {...props} />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should handle click', () => {
    // Given
    const { getByTestId } = render(<ButtonPlus {...props} />);

    // When
    fireEvent.click(getByTestId('plus-button'));

    // Then
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });

  it('should disable the click action', () => {
    // Given
    const { getByTestId } = render(<ButtonPlus {...props} disabled={true} />);

    // When
    fireEvent.click(getByTestId('plus-button'));

    // Then
    expect(props.onClick).not.toHaveBeenCalled();
  });
});
