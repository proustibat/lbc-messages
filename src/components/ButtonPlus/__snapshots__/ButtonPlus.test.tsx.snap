// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPlus should render correctly by default 1`] = `
<div>
  <button
    aria-label="add"
    class="MuiButtonBase-root MuiFab-root root_backgroundColor-#3f51b5 root_color-#fff"
    data-testid="plus-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
